src/ext/Util/ca/scaexec.cpp

```diff
@@ -715,35 +860,127 @@ static HRESULT RemoveUserInternal(
     return hr;
 }
 
-static void GetServerName(LPWSTR pwzDomain, LPWSTR* ppwzServerName)
+static HRESULT RemoveGroupInternal(
+    LPWSTR wzGroupCaData,
+    LPWSTR wzDomain,
+    LPWSTR wzName,
+    int iAttributes
+)
+{
+    HRESULT hr = S_OK;
+    UINT er = ERROR_SUCCESS;
+
+    LPWSTR pwz = NULL;
+    LPWSTR pwzGroup = NULL;
+    LPWSTR pwzGroupDomain = NULL;
+    LPCWSTR wz = NULL;
+    PDOMAIN_CONTROLLER_INFOW pDomainControllerInfo = NULL;
+
+    //
+    // Remove the Group if the group was created by us.
+    //
+    if (!(SCAG_DONT_CREATE_GROUP & iAttributes))
+    {
+        if (wzDomain && *wzDomain)
```

Comment:
This entire DsGetDcName logic should be replaced by a GetServerName function call